src/ai/backend/common/service_discovery/redis_discovery/service_discovery.py

```diff
@@ -0,0 +1,105 @@
+import json
+import uuid
+from dataclasses import dataclass
+from typing import Any, Sequence
+
+from redis.asyncio.client import Pipeline, Redis
+
+from ai.backend.common import redis_helper
+from ai.backend.common.json import dump_json_str
+from ai.backend.common.types import RedisConnectionInfo
+
+from ..service_discovery import ServiceDiscovery, ServiceMetadata
+
+_DEFAULT_PREFIX = "service_discovery"
+
+
+@dataclass
+class RedisServiceDiscoveryArgs:
+    redis: RedisConnectionInfo
+    ttl: int = 60
+    prefix: str = _DEFAULT_PREFIX
+
+
+class RedisServiceDiscovery(ServiceDiscovery):
+    _redis: RedisConnectionInfo
+    _ttl: int = 60
```

Comment:
It looks like you intended to assign a default value using `_ttl: int = 60`, but doing so here makes `_ttl` a class variable. This might be a confusion with how defaults work in `dataclass`.
Please specify default values only in the Args dataclass, not at the class level.